### ISSUE TITLE
Add search feature

### ISSUE THREAD

**Issue by xenowits** [ObolNetwork/obol-docs#53]
## Problem to be solved

Our content in the docs site is growing. Sometimes, we need to go through all the pages for the right information we want.

## Proposed solution

Add a search feature to enable searching using a keyword throughout the docs site. This can be achieved by enabling docusaurus's algolia integration, but that may require someone to open an algolia indexing backend saas account, not sure.

**Comment by OisinKyne** [comment]
Hey @EridianAlpha the hold up here is I find the NPM package for local search a little suspect. And I want the devops team to decide how comfortable we are with untrusted npm packages running in our CI/CD. Is this the only package that offers local search? And did you find it recommended from a trusted source or randomly on the internet?

**Comment by EridianAlpha** [comment]
@OisinKyne I understand. It was one of the community-supported search packages suggested by Docusaurus: https://docusaurus.io/community/resources#search 

I tried all of them and [@cmfcmf/docusaurus-search-local](https://github.com/cmfcmf/docusaurus-search-local) was the one that worked well, was customizable, had a reasonable number of stars on GitHub and appears to be actively maintained (updated just yesterday).

The other suggested options are:
[docusaurus-plugin-lunr](https://github.com/daldridge/docusaurus-plugin-lunr) - Docusaurus v2 plugin to create a local search index for use with Lunr.js
[docusaurus-lunr-search](https://github.com/lelouch77/docusaurus-lunr-search) - Offline Search for Docusaurus v2
[docusaurus-search-local](https://github.com/cmfcmf/docusaurus-search-local) - Offline/local search for Docusaurus v2
[@easyops-cn/docusaurus-search-local](https://github.com/easyops-cn/docusaurus-search-local) - Offline/local search for Docusaurus v2 (language of zh supported)
[docusaurus-theme-search-typesense](https://github.com/typesense/docusaurus-theme-search-typesense) - Docusaurus v2 plugin for [Typesense DocSearch](https://typesense.org/docs/latest/guide/docsearch.html).

**Comment by OisinKyne** [comment]
> I tried all of them and [@cmfcmf/docusaurus-search-local](https://github.com/cmfcmf/docusaurus-search-local) was the one that worked well, was customizable, had a reasonable number of stars on GitHub and appears to be actively maintained (updated just yesterday).

Then why is the one in the PR the chinese one? Can we use the cmfcmf one instead? https://github.com/ObolNetwork/obol-docs/pull/130/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R21

**Comment by OisinKyne** [comment]
If possible. I have DIYed CSS on this site too for things like the redoc api spec feature so might be able to figure it out if you get stuck trying to make cmfcmf work.

**Comment by EridianAlpha** [comment]
Ok thanks, I'll have a look and let you know, as the default styling with cmfcmf is pretty far off brand 
<img width="258" alt="image" src="https://user-images.githubusercontent.com/86061486/221692395-4c3f4dad-e0dd-4d35-a47a-dafb255f14e6.png">

**Comment by EridianAlpha** [comment]
@OisinKyne I've updated the [PR](https://github.com/ObolNetwork/obol-docs/pull/130) to use https://github.com/cmfcmf/docusaurus-search-local

I've styled it to look nicer and it seems to work well. One slight annoyance is that it uses mark.js to highlight the search term (even if that term is the letter "e") highlighting every instance of that term on the page. Maybe it's a positive thing so people can quickly find what they searched for, so it's probably just a personal preference. There's no configuration to disable it, but refreshing the page does clear the highlighting.

Deployment: https://deploy-preview-130--obol-docs.netlify.app/

<img width="270" alt="image" src="https://user-images.githubusercontent.com/86061486/221928125-4c41131a-02b0-48f2-b2d5-dcb1d5070d9c.png">
